bespin_k8s_namespace/tasks/delete_namespace.yml

```diff
@@ -0,0 +1,47 @@
+# Expects appropriate environment variables specifying ks8 configuration to already be set.
+# See https://docs.ansible.com/ansible/latest/modules/k8s_module.html
+
+- name: Remove stage system data job (to allow teardown before stage system data finishes)
+  k8s:
+    namespace: "{{ namespace_name }}"
+    state: absent
+    definition: "{{ lookup('file', 'k8s-lando-stage-system-data.yml') }}"
+
+- name: List any pods created by stage system data job
+  k8s_info:
+    namespace: "{{ namespace_name }}"
+    kind: Pod
+    label_selectors:
+      - job-name = stage-input-data
+  register: stage_system_data_pods
+
+- name: Remove stage system data pods
```

Comment:
I have a few questions about deleting the system stage data pods:

1. Since a `Job` has been created, won't it re-create `Pod`s if they get deleted through the API?
2. Why is it necessary to fetch the list of `Pod`s and delete them? Can we not simply delete the `Job`?
3. I've seen labels encouraged pretty widely for tagging resources for quick deletion. If we cannot simply delete the job by name, can we try to use that? The other advantage here is that I think we could apply the same label to different kinds of related resources (like the Pod and the PVC), and delete them all in one swoop.

Reply:
> 1. Since a Job has been created, won't it re-create Pods if they get deleted through the API?

We delete the Job just above this command. 
> Why is it necessary to fetch the list of Pods and delete them? Can we not simply delete the Job?

For some reason the pods remain after deleting the job in google cloud. If they aren't deleted the PV leaks

> I've seen labels encouraged pretty widely for tagging resources for quick deletion. 

I'll look at attaching labels to simplify the cleanup process.

Reply:
Looks like `k8s_info` makes you specify `kind` so you have to read items one kind at a time.
https://docs.ansible.com/ansible/devel/modules/k8s_info_module.html